src/main/java/org/opensearch/security/auditlog/sink/Log4JSink.java

```diff
@@ -44,9 +44,30 @@ public boolean isHandlingBackpressure() {
 
     public boolean doStore(final AuditMessage msg) {
         if(enabled) {
-            auditLogger.log(logLevel, msg.toJson());
+            logAtLevel(auditLogger, logLevel, msg.toJson());
         }
         return true;
     }
 
+    private boolean isLogLevelEnabled(Logger logger, Level level) {
```

Comment:
I just did this in a pinch to get this going... log4j has `.isEnabled()` and `.log()`, where SLF4J segregates by level. Perhaps there is a more idiomatic way of doing this.

Reply:
I don't know of a better way, but I also think this is a good solution for now. Maybe there is a more extensible solution which could come later.

You could simplify the case by returning rather than using `isEnabled` if you wanted to make the code a little tighter.